lib/koala/api/graph_batch_api.rb

```diff
@@ -41,67 +41,106 @@ def graph_call_in_batch(path, args = {}, verb = "get", options = {}, &post_proce
 
       # execute the queued batch calls
       def execute(http_options = {})
-        return [] unless batch_calls.length > 0
+        return [] if batch_calls.empty?
+
         # Turn the call args collected into what facebook expects
-        args = {}
-        args["batch"] = JSON.dump(batch_calls.map { |batch_op|
-          args.merge!(batch_op.files) if batch_op.files
-          batch_op.to_batch_params(access_token, app_secret)
-        })
+        args = { 'batch' => batch_args }
+        batch_calls.each do |call|
+          args.merge! call.files || {}
```

Comment:
I know the original uses `merge!`. I figure since we're touching it we might as well improve the code 😁 )